_posts/2020-12-24-how-to-start-with-foreman

```diff
@@ -0,0 +1,399 @@
+Select the Operating System from the drop down. Click on Generate command button and
+you will get the command that you must execute on the existing host.
+You must have `curl` installed on the machine and the command must
+be executed under root, since it may install additional packages. If your host is not configured
+to trust the Foreman CA, add `--insecure` flag to the curl command.
+Note that by this you accept the risk of man in the middle attack to your host.
+You can first try to run the curl without `| bash` part, to see the content of the script.
+After thorough investigation, you can execute it manually.
+
+If the registration was successful you should see something like
+`Successfully enrolled host host.example.com with Foreman.`. If you see this, you can
+see the Host created in your Foreman. The registration process can be tweaked, to e.g. deploy
+SSH keys, preinstall some software etc.
+
+Note the command contains JSON Web Token (JWT), that authenticates the user when curl fetches
+the registration instructions from Foreman. This token is valid for 4 hours by default.
+The token lifetime can be customized in the registration form. The same command can be used for multiple
+machines, however it also contains the selected OS id, so all such machines would be
+considered running this specific OS.
+
+### Registering hosts using subscription-manager or bootstrap.py
+
+**This section assumes you have [Katello](https://theforeman.org/plugins/katello/nightly/index.html) plugin installed.**
+
+If you use Katello, there's a way to create unmanaged host by subscribing it. This is
+specific for RPM based distributions, namely Red Hat Enterprise Linux and CentOS. The
+subscription-manager was recently ported also to DEB based distributions. However you need
+to add a custom repository to your machines in order to install it.  When
+the Host is created this way, the second object called Content Host is also automatically created.
+Content Hosts are extensions to the regular Host object, extending its. You can see that in
+Content Host UI. Note that the Host will be unmanaged, however you can convert it later to managed Host.
+Also if you provisioned the host (meaning it's managed) a Content Host can still be created.
+In fact the subscription is usually performed as part of the provisioning.
+
+Katello provides subscription management and content management features. Meaning you can control
+what product entitlements are consumed by each such Host and what repositories are available to them.
+You can control content versions, e.g. you only want the newest packages installed on your testing environment,
+while in production you want to update packages only after they were tested in the testing evironment.
+
+Based on the information we receive from katello agent, Foreman also knows, what packages
+are installed on a particular Host and whether there's some update or applicable errata. 
```

Comment:
```suggestion
Based on the information we receive from the host, Foreman also knows, what packages
are installed on a particular Host and whether there's some update or applicable errata. 
```

k-agent isn't required for errata, an k-host-tools only for systems with old sub-man (el6, el7 pre 7.6(?)) as @jturel can confirm